rs/http_utils/src/file_downloader.rs

```diff
@@ -182,16 +183,49 @@ pub fn check_file_hash(path: &Path, expected_sha256_hex: &str) -> FileDownloadRe
     }
 }
 
-/// Extract the contents of a given .tar.gz file into `target_dir`
-pub fn extract_tar_gz_into_dir(tar_gz_path: &Path, target_dir: &Path) -> FileDownloadResult<()> {
-    let map_to_untar_error = |e| FileDownloadError::untar_error(tar_gz_path, e);
+/// Check if the file is in Gzip format by verifying the first 2 bytes
+fn is_gz_file<R: Read>(reader: &mut R) -> io::Result<bool> {
+    let mut buffer = [0; 2];
+    reader.read_exact(&mut buffer)?;
+    reader.by_ref().take(2).read_to_end(&mut Vec::new())?;
```

Comment:
Why is this line needed here (and in `is_zst_file`)?

Reply:
You are correct—that line is not needed. Removed! Thank you